packages/query-core/src/queryClient.ts

```diff
@@ -118,6 +118,21 @@ export class QueryClient {
     return this.queryCache.find<TQueryFnData>(queryKey, filters)?.state.data
   }
 
+  ensureQueryData<
+    TQueryFnData = unknown,
+    TQueryKey extends QueryKey = QueryKey,
+  >(
+    queryKey: TQueryKey,
+    queryFn: QueryFunction<TQueryFnData, TQueryKey>,
+    filters?: QueryFilters,
+  ): Promise<TQueryFnData> {
```

Comment:
I think we need the same interface here as `fetchQuery`. That means:

- 3 overloads (we'll remove them in v5)
  - if you want to do me a favour, you'll write the tests with the object-only syntax :)
- the same interface as `fetchQuery`: 4 generics and `FetchQueryOptions`
- we probably don't need the `filters` on `getQueryData`. they are only needed so that you can opt-out of exact filtering and add some other fuzzy filter options instead. It doesn't really fit the use-case for `ensureQueryData`.

Reply:
Oh, I thought it was suppose to have a signature based on `getQueryData`. I'll change it, no problem.